spyder/plugins/pythonpath/container.py

```diff
@@ -74,195 +77,190 @@ def setup(self):
-    def _load_pythonpath(self):
-        """Load Python paths."""
-        # Get current system PYTHONPATH
-        system_path = get_system_pythonpath()
-
-        # Get previous system PYTHONPATH
-        previous_system_path = self.get_conf('system_path', default=())
-
-        # Load all paths
-        paths = []
-        previous_paths = self.get_conf('path')
-        for path in previous_paths:
-            # Path was removed since last time or it's not a directory
-            # anymore
-            if not osp.isdir(path):
-                continue
-
-            # Path was removed from system path
-            if path in previous_system_path and path not in system_path:
-                continue
-
-            paths.append(path)
-
-        self.path = tuple(paths)
-
-        # Update path option. This avoids loading paths that were removed in
-        # this session in later ones.
-        self.set_conf('path', self.path)
-
-        # Update system path so that path_manager_dialog can work with its
-        # latest contents.
-        self.set_conf('system_path', system_path)
-
-        # Add system path
-        if system_path:
-            self.path = self.path + system_path
-
-        # Load not active paths
-        not_active_paths = self.get_conf('not_active_path')
-        self.not_active_path = tuple(
-            name for name in not_active_paths if osp.isdir(name)
+    def _get_system_paths(self):
+        system_paths = get_system_pythonpath()
+        conf_system_paths = self.get_conf('system_paths', {})
+
+        # If a system path already exists in the configuration, use the
+        # configuration active state. If it does not exist in the
+        # configuration, then set the active state to True.
+        system_paths = OrderedDict(
+            {p: conf_system_paths.get(p, True) for p in system_paths}
         )
 
-    def _save_paths(self, new_path_dict):
-        """
-        Save tuples for all paths and not active ones to config system and
-        update their associated attributes.
+        return system_paths
 
-        `new_path_dict` is an OrderedDict that has the new paths as keys and
-        the state as values. The state is `True` for active and `False` for
-        inactive.
+    def _save_paths(self, user_paths=None, system_paths=None, prioritize=None):
         """
-        path = tuple(p for p in new_path_dict)
-        not_active_path = tuple(
-            p for p in new_path_dict if not new_path_dict[p]
-        )
-
-        # Don't set options unless necessary
-        if path != self.path:
-            self.set_conf('path', path)
-            self.path = path
+        Save user and system path dictionaries and prioritize to config.
+
+        Notes
+        -----
+        - Each dictionary key is a path and the value is the active state.
+        - sig_pythonpath_changed is emitted from this method, and nowhere else,
+        on condition that _spyder_pythonpath changed.
+
```

Comment:
```suggestion
```

Move `Notes` below `Parameters` because it's customary (I couldn't easily do that in my previous suggestion).